src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Resolvers/DynamicObjectResolver.cs

```diff
@@ -1741,7 +1741,7 @@ public static ObjectSerializationInfo CreateOrNull(Type type, bool forceStringKe
             if (ctor == null)
             {
                 ctorEnumerator =
-                    ti.DeclaredConstructors.Where(x => x.IsPublic).OrderByDescending(x => x.GetParameters().Length)
+                    ti.DeclaredConstructors.Where(x => allowPrivate ? true : x.IsPublic).OrderByDescending(x => x.GetParameters().Length)
```

Comment:
I think this reads easier

```suggestion
                    ti.DeclaredConstructors.Where(x => allowPrivate || x.IsPublic).OrderByDescending(x => x.GetParameters().Length)
```

Reply:
Sure.